python/lsst/obs/base/cameraMapper.py

```diff
@@ -1060,6 +1072,53 @@ def _resolveFilters(definitions, idFilter, filterLabel):
             matches.intersection_update(definitions.findAll(filterLabel.bandLabel))
         return matches
 
+    def _getBestFilter(self, storedLabel, idFilter):
+        """Estimate the most complete filter information consistent with the
+        file or registry.
+
+        Parameters
+        ----------
+        storedLabel : `lsst.afw.image.FilterLabel` or `None`
+            The filter previously stored in the file.
+        idFilter : `str` or `None`
+            The filter implied by the data ID, if any.
+
+        Returns
+        -------
+        bestFitler : `lsst.afw.image.FilterLabel` or `None`
+            The complete filter to describe the dataset. May be equal to
+            ``storedLabel``. `None` if no recommendation can be generated.
+        """
+        try:
+            # getGen3Instrument returns class; need to construct it.
+            filterDefinitions = self.getGen3Instrument()().filterDefinitions
+        except NotImplementedError:
+            filterDefinitions = None
+
+        if filterDefinitions is not None:
```

Comment:
There's no `else` associated with this `if`: is that intentional? This requires that there be a `FilterDefinitions` for this instrument.

Reply:
Yes, it's intentional -- if there's no `filterDefinitions`, then I don't have enough info to generate a "good" `FilterLabel`. I'll add a comment to the code.